app/src/main/java/com/xmartlabs/template/service/adapter/MillisecondsLocalDateAdapter.java

```diff
@@ -17,29 +20,25 @@
 import timber.log.Timber;
 
 /**
- * Created by mirland on 24/08/16.
+ * {@link Gson} type adapter that serializes {@link LocalDate} objects to a millisecond format
  */
 public class MillisecondsLocalDateAdapter implements JsonSerializer<LocalDate>, JsonDeserializer<LocalDate> {
   @Override
   public synchronized JsonElement serialize(LocalDate date, Type type,
                                             JsonSerializationContext jsonSerializationContext) {
-    if (date != null) {
-      return new JsonPrimitive(date.atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli());
-    }
-    return null;
+    return Optional.ofNullable(date)
+        .map(theDate -> new JsonPrimitive(theDate.atStartOfDay(ZoneOffset.UTC).toInstant().toEpochMilli()))
+        .orElse(null);
   }
 
   @Override
   public synchronized LocalDate deserialize(JsonElement jsonElement, Type type,
                                             JsonDeserializationContext jsonDeserializationContext) {
-    String dateAsString = jsonElement.getAsString();
-    if (!StringUtils.stringIsNullOrEmpty(dateAsString)) {
-      try {
-        return Instant.ofEpochMilli(Long.valueOf(dateAsString)).atZone(ZoneOffset.UTC).toLocalDate();
-      } catch (Exception e) {
-        Timber.e(e, "Date cannot be parsed, date='%s'", dateAsString);
-      }
-    }
-    return null;
+    return Optional.ofNullable(jsonElement.getAsString())
+        .filter(str -> !StringUtils.isNullOrEmpty(str))
+        .map(str -> Exceptional.of(() -> Instant.ofEpochMilli(Long.valueOf(str)).atZone(ZoneOffset.UTC).toLocalDate())
+            .ifException(e -> Timber.e(e, "Date cannot be parsed, date='%s'", str))
+            .get())
```

Comment:
What happen if the value is null? I thing using `.getOptional()` is better. Also you should change the `map` to the `flatmap`?